components/mdns/examples/query_advertise/pytest_mdns.py

```diff
@@ -64,6 +64,7 @@ def mdns_server(esp_host, events):
     sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
     sock.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
     sock.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEPORT, 1)
+    sock.setsockopt(socket.SOL_SOCKET, socket.SO_BINDTODEVICE, 'eth0\0'.encode('utf-8'))
```

Comment:
Why we are binding it to `eth0` only?

Reply:
I think the pytest script is using `eth0` to communicate with the test board on CI runner.

Reply:
If I didn't bind the interface, CI will fail https://github.com/espressif/esp-protocols/actions/runs/7550392768/job/20556119271

Reply:
There are two things concerning with this change.
1. The tests may fail if we switch to a different interface or operating system, such as macOS or Windows. I can address this issue later by updating pytest myself.
2. More worryingly, it's unclear why the tests were passing previously without being bound to the specific interface. This raises questions about potential breaking changes that could arise from this discrepancy.

Reply:
I have executed the changes without binding, in separate [PR503](https://github.com/espressif/esp-protocols/pull/503) and it passed
https://github.com/espressif/esp-protocols/actions/runs/7783718151

So I suppose this change is not necessary . Could you please, revert this and rebase with master so we can merge it?

Reply:
Reverted

Reply:
Thank you